honeybee/radiance/command/rcollate.py

```diff
@@ -13,92 +12,36 @@ class Rcollate(RadianceCommand):
-                 f=None, matrixFile=None, outputName=None):
+    def __index__(self,outputName=None,matrixFile=None,rcollateParameters=None):
+        """Init command"""
         RadianceCommand.__init__(self)
-        self.h = headerInfo
-        self.w = w
-        self.t = t
-        self.ic = ic
-        self.ir = ir
-        self.oc = oc
-        self.orX = orX
-        self.matrixFile = matrixFile
-        self.f = f
-        self.outputName = outputName
-
-    @property
-    def f(self):
-        return self._f
 
-    @f.setter
-    def f(self, value):
-        varName = "f (input format)"
-        assert value[0] in ('a', 'f', 'd', 'b'), "The first value for %s " \
-                                                 "should be either a,f,d or" \
-                                                 "b. %s was provided" \
-                                                 "insted" % (varName, value)
-        if len(value) > 1:
-            try:
-                dataRecords = int(value[1:])
-            except ValueError:
-                raise ValueError("The input for %s should be in the format"
-                                 "[afdb]N .e.g f3. The value provided was %s"
-                                 % (varName, value))
-        self._f = value
+        self.outputName = outputName
+        self.matrixFile = matrixFile
+        self.rcollateParameters = rcollateParameters
 
     def toRadString(self, relativePath=False):
-        """Return full command as a string."""
-        fmtBool = Rfmt.boolean
-        fmtJoin = Rfmt.joined
-        fmtNormal = Rfmt.normal
-
-        fmtDict = {'h': ('h', fmtJoin),
-                   'w': ('w', fmtBool),
-                   't': ('t', fmtBool),
-                   'ic': ('ic', fmtNormal),
-                   'ir': ('ir', fmtNormal),
-                   'oc': ('oc', fmtNormal),
-                   'orX': ('or', fmtNormal),
-                   'f': ('f', fmtJoin)}
 
-        commandInputs = ""
-        for key, value in fmtDict.items():
-                currentValue = getattr(self, key)
-                if currentValue is not None:
-                    currentFlag, currentFormatter = value
-                    commandInputs += currentFormatter(currentFlag, currentValue)
+        outputFile = os.path.splitext(str(self.matrixFile))[0] + ".mtx" \
+            if self.outputName is None and self.matrixFile.normpath is not None \
+            else self.outputName
 
-        outputFileName = self.outputName
-        if not self.outputName:
-            try:
-                outputFileName = os.path.splitext(self.inputFiles[0])[0] + \
-                                 '.dat'
-            except TypeError:
-                outputFileName = None
 
         radString = "%s %s %s > %s" % (
             os.path.join(self.radbinPath, 'rcollate'),
-            commandInputs,
-            self.inputFiles,
-            outputFileName
+            self.rcollateParameters.toRadString(),
+            self.matrixFile,
```

Comment:
@sariths, This is quite new but all path files should be passed to ```self.normspace``` in radToString methods in order to be able to handle file path with white-space (e.g. self.normspace(self.matrixFile)).